lib/DriverTool/autolink_extract_main.cpp

```diff
@@ -164,8 +176,18 @@ extractLinkerFlagsFromObjectFile(const llvm::object::WasmObjectFile *ObjectFile,
       llvm::SmallVector<llvm::StringRef, 4> SplitFlags;
       SegmentData.split(SplitFlags, llvm::StringRef("\0", 1), -1,
                         /*KeepEmpty=*/false);
-      for (const auto &Flag : SplitFlags)
-        LinkerFlags.push_back(Flag.str());
+      for (const auto &Flag : SplitFlags) {
+        // If it is one of the common Swift library flags,
+        // only add if not seen before.
+        if (auto It = CommonAddedSwiftLibraries.find(Flag.str());
+            It != CommonAddedSwiftLibraries.end()) {
+          if (It->second == false) {
+            LinkerFlags.push_back(Flag.str());
+            It->second = true;
+          }
+        } else
+          LinkerFlags.push_back(Flag.str());
```

Comment:
I think that this might be more concise as:

```c++
for (const auto &Flag : SplitFlags)
  if (SwiftRuntimes.find(Flag.str()) == SwiftRuntimes.end())
    LinkerFlags.emplace_back(Flag.str());
  else
    entry->second = true;
for (const auto &Runtime : SwiftRuntimes)
  if (Runtime->second)
    LinkerFlags.push_back(Runtime->first);
```

We already know that the runtimes should be the very last set of dependencies, so we should delay it to the end.

Reply:
I like the suggestion, but I think the second loop here should only run at the very end when emitting linker flags, not when adding them here per-object-file because then we'd still have duplicate entries.